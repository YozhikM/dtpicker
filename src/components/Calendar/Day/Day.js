/* @flow */

import React from 'react';
import { getDate, isWithinRange, isSameDay } from 'date-fns';

type Props = {
  date: Date,
  isActive: boolean,
  onClick: Function,
  startDate: any,
  endDate: any
};

type State = void;

class Day extends React.Component<Props, State> {
  onClick = () => {
    const { onClick, date } = this.props;
    if (onClick) {
      onClick(date);
    }
  };

  filterRangeDay = (date: Date, startDate: Date, endDate: Date) => {
    let result;
    if (startDate < endDate) {
      result = isWithinRange(date, startDate, endDate);
    } else {
      result = isWithinRange(date, endDate, startDate);
    }
    return result;
  };

  checkStartDate = (date: Date, startDate: Date) => {
    let result;
    if (isSameDay(startDate, date)) {
      result = true;
    } else {
      result = false;
    }
    return result;
  };

  render() {
    const { date, isActive, startDate, endDate } = this.props;
    let inRange = this.filterRangeDay(date, startDate, endDate);
    let startDay = this.checkStartDate(date, startDate);
    const style = {
      backgroundColor: (isActive || startDay ? '#34495e' : null) || (inRange ? '#9bb0c5' : null),
      color: isActive || inRange || startDay ? 'white' : '#34495e',
      border: isActive || inRange || startDay ? '1px solid transparent' : '1px solid #e4e7e7'
    };
    let dayOfMonth = getDate(date);
    return (
      <td style={style} onClick={this.onClick}>
        {dayOfMonth}
      </td>
    );
  }
}

export default Day;
