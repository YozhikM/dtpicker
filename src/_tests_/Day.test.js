import React from 'react';
import Day from '../components/Calendar/Day/Day';
import renderer from 'react-test-renderer';

it ('renders correctly', () => {
  const tree = renderer.create(
    <Day date={new Date()}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
