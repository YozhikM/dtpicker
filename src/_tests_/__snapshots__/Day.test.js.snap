// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<td
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": null,
      "border": "1px solid #e4e7e7",
      "color": "#34495e",
    }
  }
>
  5
</td>
`;
